### ISSUE TITLE
Wrong Retention Policy in the UI

### ISSUE THREAD

**Issue by modem** [garethgeorge/backrest#74]
Hi, I just noticed the plans retention policy is not showing properly in the UI.
I've set the plan to keep the last 7 backups, but the UI is showing none:
![image](https://github.com/garethgeorge/backrest/assets/62859/a6eeed4b-4127-45bf-885b-48e47831411a)

Selecting retention by count defaults to 30:
![image](https://github.com/garethgeorge/backrest/assets/62859/4a03db7e-14de-4f3c-893f-f47112cacb3c)

The config.json file shows the correct configuration:
![image](https://github.com/garethgeorge/backrest/assets/62859/43ea6cb1-479b-46e7-bb9a-6109c0bc8e5b)

I also tested setting the retention by time, and the same behavior happens, but the config file is correctly saved.

**Comment by SiddharthManthan** [comment]
I created a backup with retention policy set to none. I then modified it to "By Time Period" but the changes are not applied. config.json has been modified but backrest does not display it in ui, it still displays none. During backup it logs error that backup retention has not been configured.

**Comment by modem** [comment]
This is odd, I just ran a backup in 0.11.1 and the Forget task run well. 
![image](https://github.com/garethgeorge/backrest/assets/62859/f9c4d653-b952-430f-b3fd-0ed0b8ba8f15)
But for me it's configured to count = 7. Maybe there's a different behavior when we configure by Time Period.

**Comment by garethgeorge** [comment]
Thanks both for flagging this, I'm seeing the regression as well. I significantly reworked the way the UI handles form data in the last update but it looks like I broke something here. 

I'll get out a fix to the UI regression (hopefully that's the last of the major bug patch releases needed -- the changes in 0.11.* have been significant so there are definitely a few bugs that slipped in!).

What error are you seeing w.r.t. retention not configured? I wouldn't expect anything to print to stdout.

**Comment by SiddharthManthan** [comment]
It does get configured but it gets overwritten to null (open and close that screen a few times ?).


________________________________

**Comment by modem** [comment]
> Thanks both for flagging this, I'm seeing the regression as well. I significantly reworked the way the UI handles form data in the last update but it looks like I broke something here.
> 
> I'll get out a fix to the UI regression (hopefully that's the last of the major bug patch releases needed -- the changes in 0.11.* have been significant so there are definitely a few bugs that slipped in!).
> 
> What error are you seeing w.r.t. retention not configured? Is it coming from the config update e.g. `invalid retention policy: ...`?

Not sure the question was for me or not.
With my policy set to 7 (count), I have no error, the forget task runs as expected:
I edited the plan and saved, so the retention is saved set to none (null in the config file). When I run the plan, I also get no error, but of course there is no snapshot deletion task triggered.

**Comment by modem** [comment]
> It does get configured but it gets overwritten to null (open and close that screen a few times ?).
I don't see it change if I open and close the plan configuration window a few times, unless I click on **Save**, instead of **Cancel**

**Comment by garethgeorge** [comment]
To clarify on the bug as I understand it, it's coming from some quirks of the UI framework I use (antd) and it's handling of a form.

When you first configure the retention policy and submit the form, the info is correctly captured and submitted with the form. When re-opening the edit view, the UI isn't correctly rehydrating the state of the form. B/c the existing state isn't picked up, the mode defaults to "none" and the existing values actually get overwritten and cleared.

From here there are two bugs that can happen

 1. The user does not make any changes to the retention policy. In this case, the value becomes "null" and is the retention policy is cleared if the form is resubmitted.
 2. The user clicks any of the retention policy modes and then sets the form back to "null". In this (and possibly some other strange cases) the retention policy can become empty. An empty retention policy is invalid (it must either be null or specify some of it's fields e.g. last n or time based). In this case I think you may also see some weird validation errors when trying to submit the form.

Tl;dr though, retention policy still works and executes if you see it in your config. This is a UI only bug with a few subtle behaviors. There isn't any case where it should be possible to inadvertently set a retention policy that looses data (though it's definitely hard to tell what you've set the policy to without viewing the config).

**Comment by garethgeorge** [comment]
Fixed in latest release.
